server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -66,50 +69,144 @@
         Property.NodeScope
     );
 
-    public static final ByteSizeValue DEFAULT_MAX_BYTES_PER_SEC = new ByteSizeValue(40L, ByteSizeUnit.MB);
+    /**
+     * Disk's write bandwidth allocated for this node. This bandwidth is expressed for write operations that have the default block size of
+     * {@link #DEFAULT_CHUNK_SIZE}.
+     */
+    public static final Setting<ByteSizeValue> NODE_DISK_AVAILABLE_WRITE_BANDWIDTH_SETTING = bandwidthSetting(
+        "node.disk.allocated_write_bandwidth.default"
+    );
 
-    public static final Setting<ByteSizeValue> INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING = Setting.byteSizeSetting(
-        "indices.recovery.max_bytes_per_sec",
-        s -> {
-            final List<DiscoveryNodeRole> roles = NodeRoleSettings.NODE_ROLES_SETTING.get(s);
-            final List<DiscoveryNodeRole> dataRoles = roles.stream().filter(DiscoveryNodeRole::canContainData).toList();
-            if (dataRoles.isEmpty()) {
-                // if the node is not a data node, this value doesn't matter, use the default
-                return DEFAULT_MAX_BYTES_PER_SEC.getStringRep();
+    /**
+     * Disk's read bandwidth allocated for this node. This bandwidth is expressed for read operations that have the default block size of
+     * {@link #DEFAULT_CHUNK_SIZE}.
+     */
+    public static final Setting<ByteSizeValue> NODE_DISK_AVAILABLE_READ_BANDWIDTH_SETTING = bandwidthSetting(
+        "node.disk.allocated_read_bandwidth.default"
+    );
+
+    /**
+     * Network's read bandwidth allocated for this node.
+     */
+    public static final Setting<ByteSizeValue> NODE_NETWORK_AVAILABLE_BANDWIDTH_SETTING = bandwidthSetting(
+        "node.network.allocated_bandwidth"
+    );
+
+    static final double DEFAULT_FACTOR_VALUE = 0.8d;
```

Comment:
I think we should lower this to say 0.4. 0.8 seems like it could disturb ongoing operations?

Reply:
I agree, I set it to `0.4` in d9ac1485c7605e0378bf1b3db1910107a8a55fc1.